### PR TITLE
Adding get_default method to the projects endpoint

### PR DESCRIPTION
**Pull request by LGraber**
Want a quick view on this. I sent Russell a mail cause our endpoints are broken because of the default sorting that was added. Ignoring that, I added support for get_default and put an implementation for older versions so that our samples could be updated, too. I accidentally parented off of builders so I probably need to reparent. Tests are still to be done

### CODE REVIEW

File: samples/options_builder.py [hunk 1]
```diff
@@ -1,4 +1,5 @@
 import sys
```
Comment (t8y8) [inline samples/options_builder.py]: This file shouldn't be in the commit for this, right? I think it's part of Options Builder

File: samples/publish_workbook.py [hunk 2]
```diff
@@ -41,14 +41,13 @@ def main():
     # Step 1: Sign in to server.
     tableau_auth = TSC.TableauAuth(args.username, password)
     server = TSC.Server(args.server)
+    server.use_server_version()
```
Comment (t8y8) [inline samples/publish_workbook.py]: Thanks :)

File: tableauserverclient/server/endpoint/projects_endpoint.py [hunk 3]
```diff
@@ -20,6 +25,20 @@ def get(self, req_options=None):
         all_project_items = ProjectItem.from_response(server_response.content)
         return all_project_items, pagination_item
 
+    def get_default(self):
+        if (Version(self.parent_srv.version) < Version("2.7")):
+            for proj in Pager(self):
```
Comment (t8y8) [inline tableauserverclient/server/endpoint/projects_endpoint.py]: Nit `Pager(self.parent_srv.projects)` is a little clearer than remembering what `self` is pointing to

File: tableauserverclient/server/endpoint/projects_endpoint.py [hunk 4]
```diff
@@ -20,6 +25,20 @@ def get(self, req_options=None):
...
+                if proj.is_default():
+                    return proj
+
+            return None
```
Comment (t8y8) [inline tableauserverclient/server/endpoint/projects_endpoint.py]: You can nix this, `None` is always returned implicitly.

File: tableauserverclient/server/endpoint/projects_endpoint.py [hunk 5]
```diff
@@ -20,6 +25,20 @@ def get(self, req_options=None):
...
+        else:
+            ro = RequestOptions()
+            ro.filter.add(Filter("name", "eq", "Default"))
```
Comment (t8y8) [inline tableauserverclient/server/endpoint/projects_endpoint.py]: Sometimes it's uppercase and sometimes it's lower case (I think pre 9.0 installs vs post) so you'll need to normalize the casing

File: tableauserverclient/server/endpoint/projects_endpoint.py [hunk 6]
```diff
@@ -20,6 +25,20 @@ def get(self, req_options=None):
...
+            items, _ = self.get(ro)
+            return items[0] if (len(items) > 0) else None
```
Comment (t8y8) [inline tableauserverclient/server/endpoint/projects_endpoint.py]: `return items.pop() if items else None`

File: tableauserverclient/server/endpoint/projects_endpoint.py (hunk 5 above)
Comment (LGraber) [inline tableauserverclient/server/endpoint/projects_endpoint.py]: I don't think we support pre-9.0 anymore so I am okay with that. :)

File: tableauserverclient/server/endpoint/projects_endpoint.py (hunk 6 above)
Comment (LGraber) [inline tableauserverclient/server/endpoint/projects_endpoint.py]: ok

**Comment by LGraber** [comment]
Thanks for all the feedback

File: tableauserverclient/server/endpoint/projects_endpoint.py (hunk 5 above)
Comment (t8y8) [inline tableauserverclient/server/endpoint/projects_endpoint.py]: No, like, if their Server was upgraded from pre 9.0.

You must do this!

**Comment by RussTheAerialist** [comment]
Closing for inactivity

**Pull request closed without merging**